File: 812-rotate-string/rotate-string.py
class Solution(object):
    def rotateString(self, s, goal):
        if len(s)!=len(goal):
            return False
        for i in range(len(goal)):
            if s[0]==goal[i]:
                k=(i+1)%len(goal)
                count=0
                for j in range(1,len(goal)):
                    if s[j]!=goal[k]:
                        break
                    k=(k+1)%len(goal)
                    count+=1
                if len(goal)==count+1:
                    return True
        return False




        

File: 812-rotate-string/rotate-string.java
class Solution {
    public boolean rotateString(String s, String goal) {
        if(s.length()!=goal.length())
        return false;
        int i=0,j=0,k=0;
        while(i<s.length())
        {
            if(s.charAt(i)==goal.charAt(0))
            {
                j=i;k=0;
                while(k<s.length() &&s.charAt(j%s.length())==goal.charAt(k))
                {
                    k++;
                    j=(j+1)%s.length();
                }
                if(k==s.length())
                return true;
            }
            i++;
        }
        return false;
    }
}